server/controllers/admin/suppliers.js

```diff
@@ -0,0 +1,132 @@
+/**
+* Supplier Controller
+*
+* This controller exposes an API to the client for reading and writing Supplier
+
+*/
+var db = require('../../lib/db');
+var uuid = require('../../lib/guid');
+// GET /Supplier
+//
+// The Supplier  is assumed from the session.
+exports.list = function list(req, res, next) {
```

Comment:
The list should only send back `uuid` and `name`.

You can also write this SQL much better.  Why not do something like this:
```js
 if (req.query.locked === 0) {
   sql += 'WHERE supplier.locked = 0;';
}
```
It is much more maintainable the rewriting the entire query three times.  What happens if we add another column to the table?  We will have to make 3 replacements instead of 1.